### ISSUE TITLE
UC_ERR_WRITE_UNMAPPED error on ARMv5 binary

### ISSUE THREAD

**Issue by hackingump** [qilingframework/qiling#414]
```python

In [5]: ql = qiling.Qiling(["hello_world_arm"], rootfs=".")                                                                                                                                                                                                                    

In [6]: ql.run()                                                                                                                                                                                                                                                               
brk(0x0)
brk(0x8ed08)
settls(0x8e4c0)
uname(0x7ff3cc70) = 0
readlink(/proc/self/exe, 0x7ff3bd80, 0x1000) = -1
brk(0xb0000)
mprotect(0x89000, 0x1000, 0x1) = 0
[!] Emulation Error

[-] r0	:	 0x1
[-] r1	:	 0x7ff3cf44
[-] r2	:	 0x7ff3cf4c
[-] r3	:	 0x7fa3ce00
[-] r4	:	 0x10cd0
[-] r5	:	 0x10158
[-] r6	:	 0x8b3f8
[-] r7	:	 0x10158
[-] r8	:	 0x0
[-] r9	:	 0x0
[-] r10	:	 0x0
[-] r11	:	 0x7ff3ce04
[-] r12	:	 0x7ff3ce18
[-] sp	:	 0x7fa3cdf0
[-] lr	:	 0x107f8
[-] pc	:	 0x10524
[-] cpsr	:	 0x600001d3
[-] c1_c0_2	:	 0xf00000
[-] c13_c0_3	:	 0x8e4c0
[-] fpexc	:	 0x40000000

[+] PC = 0x10524
[+] Start      End        Perm.  Path
[+] 00010000 - 00079000 - r-x    /home/zorro/Desktop/QilingUnpacker/Qiling-Tests/hello_world_arm
[+] 00089000 - 0008b000 - rw-    /home/zorro/Desktop/QilingUnpacker/Qiling-Tests/hello_world_arm
[+] 0008b000 - 0008c000 - rwx    /home/zorro/Desktop/QilingUnpacker/Qiling-Tests/hello_world_arm
[+] 0008c000 - 0008e000 - rwx    hook mem
[+] 0008e000 - 0008f000 - rwx    [brk]
[+] 0008f000 - 000b0000 - rwx    [brk]
[+] 7ff0d000 - 7ff3d000 - rwx    [stack]
[+] ffff0000 - ffff1000 - rwx    [arm_tls]
[+] ['0x0', '0x48', '0x2d', '0xe9', '0x4', '0xb0', '0x8d', '0xe2']

[+] 0x10524        00 48 2d e9 04 b0 8d e2 05 d6 4d e2 10 d0 4d e2 05 36 4b e2 04 30 43 e2 0c 00 03 e5 05 36 4b e2 04 30 43 e2 10 10 03 e5 38 30 9f e5 00 30 93 e5 08 30 0b e5 00 30 a0 e3 2c 00 9f e5 9b 1c 00 eb push {fp, lr}
add fp, sp, #4
sub sp, sp, #0x500000
sub sp, sp, #0x10
sub r3, fp, #0x500000
sub r3, r3, #4
str r0, [r3, #-0xc]
sub r3, fp, #0x500000
sub r3, r3, #4
str r1, [r3, #-0x10]
ldr r3, [pc, #0x38]
ldr r3, [r3]
str r3, [fp, #-8]
mov r3, #0
ldr r0, [pc, #0x2c]
bl #0x177d4
---------------------------------------------------------------------------
UcError                                   Traceback (most recent call last)
<ipython-input-6-0148c347463d> in <module>
----> 1 ql.run()

/usr/local/lib/python3.8/dist-packages/qiling-1.1.dev0-py3.8.egg/qiling/core.py in run(self, begin, end, timeout, count)
    196 
    197         # emulate the binary
--> 198         self.os.run()
    199 
    200         # resume with debugger

/usr/local/lib/python3.8/dist-packages/qiling-1.1.dev0-py3.8.egg/qiling/os/linux/linux.py in run(self)
    120                         self.ql.loader.elf_entry = self.ql.entry_point
    121 
--> 122                     self.ql.emu_start(self.ql.loader.elf_entry, self.exit_point, self.ql.timeout, self.ql.count)
    123 
    124         except:

/usr/local/lib/python3.8/dist-packages/qiling-1.1.dev0-py3.8.egg/qiling/core.py in emu_start(self, begin, end, timeout, count)
    255     # start emulation
    256     def emu_start(self, begin, end, timeout=0, count=0):
--> 257         self.uc.emu_start(begin, end, timeout, count)

~/.local/lib/python3.8/site-packages/unicorn/unicorn.py in emu_start(self, begin, until, timeout, count)
    315         status = _uc.uc_emu_start(self._uch, begin, until, timeout, count)
    316         if status != uc.UC_ERR_OK:
--> 317             raise UcError(status)
    318 
    319     # stop emulation

UcError: Invalid memory write (UC_ERR_WRITE_UNMAPPED)

```

```bash
zorro@zorro-VirtualBox:~/Desktop/QilingUnpacker/Qiling-Tests$ file hello_world_arm 
hello_world_arm: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, BuildID[sha1]=a67968fcaa3bad58a808e157fd3b6a33078902b2, for GNU/Linux 3.2.0, not stripped
```

Happens on upx packed and non packed samples

**Comment by xwings** [comment]
clearly not enuf information. error msg is very generic. 

what you can do is trace the issue and try to play around with some set_syscall or set_api. Hopefully to see some PR from you,

**Comment by xwings** [comment]
We made some changes in arm and how it load libc. will you be able to try ?

**Comment by bkerler** [comment]
This might be a bug which could be solved with PR #577

**Comment by xwings** [comment]
closed
